diff --git a/src/ffi.rs b/src/ffi.rs
index 13d98cd..39564dc 100644
--- a/src/ffi.rs
+++ b/src/ffi.rs
@@ -42,7 +42,10 @@ use crate::*;
 
 #[no_mangle]
 pub extern fn quiche_version() -> *const u8 {
-    static VERSION: &str = concat!(env!("CARGO_PKG_VERSION"), "\0");
+    //static VERSION: &str = concat!(env!("CARGO_PKG_VERSION"), "\0");
+    // ANDROID's build system doesn't support environment variables
+    // so we hardcode the package version here.
+    static VERSION: &str = concat!("0.6.0", "\0");
     VERSION.as_ptr()
 }
 
