From 54e3327b143762d8c2ca8e7ff22355429492428a Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Tue, 19 May 2020 17:21:42 -0600
Subject: [PATCH 2/7] Add xxxhdpi icon_webview.png to BUILD.gn

---
 android_webview/nonembedded/BUILD.gn | 1 +
 1 file changed, 1 insertion(+)

diff --git a/android_webview/nonembedded/BUILD.gn b/android_webview/nonembedded/BUILD.gn
index 09eed4b0aed2f..38903a3cdd4d6 100644
--- a/android_webview/nonembedded/BUILD.gn
+++ b/android_webview/nonembedded/BUILD.gn
@@ -223,6 +223,7 @@ android_resources("icon_resources") {
     "java/res_icon/drawable-mdpi/icon_webview.png",
     "java/res_icon/drawable-xhdpi/icon_webview.png",
     "java/res_icon/drawable-xxhdpi/icon_webview.png",
+    "java/res_icon/drawable-xxxhdpi/icon_webview.png",
   ]
 }
 
-- 
2.42.1

