diff --git a/src/lib.rs b/src/lib.rs
index dc0699c..bd4e4fa 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -5,7 +5,10 @@
 #![allow(non_upper_case_globals)]
 #[allow(clippy::all)]
 mod bindings {
-    include!(env!("BINDING_PATH"));
+    //include!(env!("BINDING_PATH"));
+    // ANDROID's build system doesn't support environment variables
+    // so we hardcode the output location of the bindings here.
+    include!(concat!(env!("OUT_DIR"), "/grpc-bindings.rs"));
 }
 mod grpc_wrap;
 
