diff --git a/protos/Android.bp b/protos/Android.bp
index eaf06cf1..32a8f81b 100644
--- a/protos/Android.bp
+++ b/protos/Android.bp
@@ -17,7 +17,9 @@ rust_library {
     crate_name: "protos",
     srcs: ["src/lib.rs"],
     edition: "2018",
+    features: ["composite-disk"],
     rustlibs: [
+        "libcdisk_spec_proto",
         "libprotobuf",
     ],
 }
@@ -30,7 +32,9 @@ rust_defaults {
     test_suites: ["general-tests"],
     auto_gen_config: true,
     edition: "2018",
+    features: ["composite-disk"],
     rustlibs: [
+        "libcdisk_spec_proto",
         "libprotobuf",
     ],
 }
