diff --git a/Android.bp b/Android.bp
index 6b8ca5b..222916b 100644
--- a/Android.bp
+++ b/Android.bp
@@ -50,6 +50,11 @@ rust_library {
         "libpin_project_lite",
     ],
     proc_macros: ["libtokio_macros"],
+    apex_available: [
+        "//apex_available:platform",
+        "com.android.resolv",
+    ],
+    min_sdk_version: "29",
 }
 
 rust_defaults {
@@ -61,6 +66,7 @@ rust_defaults {
     features: [
         "bytes",
         "fs",
+        "full",
         "io-util",
         "libc",
         "macros",
@@ -108,36 +114,6 @@ rust_test {
     srcs: ["tests/buffered.rs"],
 }
 
-rust_test_host {
-    name: "tokio_host_test_tests_fs_file",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/fs_file.rs"],
-    test_options: {
-        unit_test: true,
-    },
-}
-
-rust_test {
-    name: "tokio_device_test_tests_fs_file",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/fs_file.rs"],
-}
-
-rust_test_host {
-    name: "tokio_host_test_tests_fs_link",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/fs_link.rs"],
-    test_options: {
-        unit_test: true,
-    },
-}
-
-rust_test {
-    name: "tokio_device_test_tests_fs_link",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/fs_link.rs"],
-}
-
 rust_test_host {
     name: "tokio_host_test_tests_io_async_read",
     defaults: ["tokio_defaults"],
@@ -348,51 +324,6 @@ rust_test {
     srcs: ["tests/no_rt.rs"],
 }
 
-rust_test_host {
-    name: "tokio_host_test_tests_process_issue_2174",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/process_issue_2174.rs"],
-    test_options: {
-        unit_test: true,
-    },
-}
-
-rust_test {
-    name: "tokio_device_test_tests_process_issue_2174",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/process_issue_2174.rs"],
-}
-
-rust_test_host {
-    name: "tokio_host_test_tests_process_issue_42",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/process_issue_42.rs"],
-    test_options: {
-        unit_test: true,
-    },
-}
-
-rust_test {
-    name: "tokio_device_test_tests_process_issue_42",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/process_issue_42.rs"],
-}
-
-rust_test_host {
-    name: "tokio_host_test_tests_process_smoke",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/process_smoke.rs"],
-    test_options: {
-        unit_test: true,
-    },
-}
-
-rust_test {
-    name: "tokio_device_test_tests_process_smoke",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/process_smoke.rs"],
-}
-
 rust_test_host {
     name: "tokio_host_test_tests_rt_basic",
     defaults: ["tokio_defaults"],
@@ -423,111 +354,6 @@ rust_test {
     srcs: ["tests/rt_threaded.rs"],
 }
 
-rust_test_host {
-    name: "tokio_host_test_tests_signal_ctrl_c",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/signal_ctrl_c.rs"],
-    test_options: {
-        unit_test: true,
-    },
-}
-
-rust_test {
-    name: "tokio_device_test_tests_signal_ctrl_c",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/signal_ctrl_c.rs"],
-}
-
-rust_test_host {
-    name: "tokio_host_test_tests_signal_drop_rt",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/signal_drop_rt.rs"],
-    test_options: {
-        unit_test: true,
-    },
-}
-
-rust_test {
-    name: "tokio_device_test_tests_signal_drop_rt",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/signal_drop_rt.rs"],
-}
-
-rust_test_host {
-    name: "tokio_host_test_tests_signal_drop_signal",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/signal_drop_signal.rs"],
-    test_options: {
-        unit_test: true,
-    },
-}
-
-rust_test {
-    name: "tokio_device_test_tests_signal_drop_signal",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/signal_drop_signal.rs"],
-}
-
-rust_test_host {
-    name: "tokio_host_test_tests_signal_multi_rt",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/signal_multi_rt.rs"],
-    test_options: {
-        unit_test: true,
-    },
-}
-
-rust_test {
-    name: "tokio_device_test_tests_signal_multi_rt",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/signal_multi_rt.rs"],
-}
-
-rust_test_host {
-    name: "tokio_host_test_tests_signal_no_rt",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/signal_no_rt.rs"],
-    test_options: {
-        unit_test: true,
-    },
-}
-
-rust_test {
-    name: "tokio_device_test_tests_signal_no_rt",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/signal_no_rt.rs"],
-}
-
-rust_test_host {
-    name: "tokio_host_test_tests_signal_notify_both",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/signal_notify_both.rs"],
-    test_options: {
-        unit_test: true,
-    },
-}
-
-rust_test {
-    name: "tokio_device_test_tests_signal_notify_both",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/signal_notify_both.rs"],
-}
-
-rust_test_host {
-    name: "tokio_host_test_tests_signal_twice",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/signal_twice.rs"],
-    test_options: {
-        unit_test: true,
-    },
-}
-
-rust_test {
-    name: "tokio_device_test_tests_signal_twice",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/signal_twice.rs"],
-}
-
 rust_test_host {
     name: "tokio_host_test_tests_sync_barrier",
     defaults: ["tokio_defaults"],
@@ -603,21 +429,6 @@ rust_test {
     srcs: ["tests/sync_mutex_owned.rs"],
 }
 
-rust_test_host {
-    name: "tokio_host_test_tests_sync_once_cell",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/sync_once_cell.rs"],
-    test_options: {
-        unit_test: true,
-    },
-}
-
-rust_test {
-    name: "tokio_device_test_tests_sync_once_cell",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/sync_once_cell.rs"],
-}
-
 rust_test_host {
     name: "tokio_host_test_tests_sync_rwlock",
     defaults: ["tokio_defaults"],
@@ -738,21 +549,6 @@ rust_test {
     srcs: ["tests/tcp_shutdown.rs"],
 }
 
-rust_test_host {
-    name: "tokio_host_test_tests_time_interval",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/time_interval.rs"],
-    test_options: {
-        unit_test: true,
-    },
-}
-
-rust_test {
-    name: "tokio_device_test_tests_time_interval",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/time_interval.rs"],
-}
-
 rust_test_host {
     name: "tokio_host_test_tests_time_rt",
     defaults: ["tokio_defaults"],
@@ -768,21 +564,6 @@ rust_test {
     srcs: ["tests/time_rt.rs"],
 }
 
-rust_test_host {
-    name: "tokio_host_test_tests_time_timeout",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/time_timeout.rs"],
-    test_options: {
-        unit_test: true,
-    },
-}
-
-rust_test {
-    name: "tokio_device_test_tests_time_timeout",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/time_timeout.rs"],
-}
-
 rust_test_host {
     name: "tokio_host_test_tests_uds_split",
     defaults: ["tokio_defaults"],
@@ -797,18 +578,3 @@ rust_test {
     defaults: ["tokio_defaults"],
     srcs: ["tests/uds_split.rs"],
 }
-
-rust_test_host {
-    name: "tokio_host_test_tests_uds_stream",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/uds_stream.rs"],
-    test_options: {
-        unit_test: true,
-    },
-}
-
-rust_test {
-    name: "tokio_device_test_tests_uds_stream",
-    defaults: ["tokio_defaults"],
-    srcs: ["tests/uds_stream.rs"],
-}
