--- libicu/ndk_headers/unicode/ulocdata.h	2021-02-10 23:39:51.609654012 +0000
+++ libicu/ndk_headers/unicode/ulocdata.h	2021-02-10 23:39:51.837655519 +0000
@@ -21,87 +21,11 @@
 
 #include "unicode/uloc.h"
 
-#if U_SHOW_CPLUSPLUS_API
-#include "unicode/localpointer.h"
-#endif   // U_SHOW_CPLUSPLUS_API
-
-/**
- * \file
- * \brief C API: Provides access to locale data.
- */
-
-/** Forward declaration of the ULocaleData structure. @stable ICU 3.6 */
-struct ULocaleData;
-
-/** A locale data object. @stable ICU 3.6 */
-typedef struct ULocaleData ULocaleData;
-
-
-
-/** The possible types of exemplar character sets.
-  * @stable ICU 3.4
-  */
-typedef enum ULocaleDataExemplarSetType  {
-    /** Basic set @stable ICU 3.4 */
-    ULOCDATA_ES_STANDARD=0,
-    /** Auxiliary set @stable ICU 3.4 */
-    ULOCDATA_ES_AUXILIARY=1,
-    /** Index Character set @stable ICU 4.8 */
-    ULOCDATA_ES_INDEX=2,
-    /** Punctuation set @stable ICU 51 */
-    ULOCDATA_ES_PUNCTUATION=3,
-#ifndef U_HIDE_DEPRECATED_API
-    /**
-     * One more than the highest normal ULocaleDataExemplarSetType value.
-     * @deprecated ICU 58 The numeric value may change over time, see ICU ticket #12420.
-     */
-    ULOCDATA_ES_COUNT=4
-#endif  /* U_HIDE_DEPRECATED_API */
-} ULocaleDataExemplarSetType;
-
-/** The possible types of delimiters.
-  * @stable ICU 3.4
-  */
-typedef enum ULocaleDataDelimiterType {
-    /** Quotation start @stable ICU 3.4 */
-    ULOCDATA_QUOTATION_START = 0,
-    /** Quotation end @stable ICU 3.4 */
-    ULOCDATA_QUOTATION_END = 1,
-    /** Alternate quotation start @stable ICU 3.4 */
-    ULOCDATA_ALT_QUOTATION_START = 2,
-    /** Alternate quotation end @stable ICU 3.4 */
-    ULOCDATA_ALT_QUOTATION_END = 3,
-#ifndef U_HIDE_DEPRECATED_API
-    /**
-     * One more than the highest normal ULocaleDataDelimiterType value.
-     * @deprecated ICU 58 The numeric value may change over time, see ICU ticket #12420.
-     */
-    ULOCDATA_DELIMITER_COUNT = 4
-#endif  /* U_HIDE_DEPRECATED_API */
-} ULocaleDataDelimiterType;
 
 
 
 
 
-#if U_SHOW_CPLUSPLUS_API
-
-U_NAMESPACE_BEGIN
-
-/**
- * \class LocalULocaleDataPointer
- * "Smart pointer" class, closes a ULocaleData via ulocdata_close().
- * For most methods see the LocalPointerBase base class.
- *
- * @see LocalPointerBase
- * @see LocalPointer
- * @stable ICU 4.4
- */
-U_DEFINE_LOCAL_OPEN_POINTER(LocalULocaleDataPointer, ULocaleData, ulocdata_close);
-
-U_NAMESPACE_END
-
-#endif
 
 
 
@@ -111,22 +35,6 @@
 
 
 
-/**
- * Enumeration for representing the measurement systems.
- * @stable ICU 2.8
- */
-typedef enum UMeasurementSystem {
-    UMS_SI,     /**< Measurement system specified by SI otherwise known as Metric system. @stable ICU 2.8 */
-    UMS_US,     /**< Measurement system followed in the United States of America. @stable ICU 2.8 */
-    UMS_UK,     /**< Mix of metric and imperial units used in Great Britain. @stable ICU 55 */
-#ifndef U_HIDE_DEPRECATED_API
-    /**
-     * One more than the highest normal UMeasurementSystem value.
-     * @deprecated ICU 58 The numeric value may change over time, see ICU ticket #12420.
-     */
-    UMS_LIMIT
-#endif  /* U_HIDE_DEPRECATED_API */
-} UMeasurementSystem;
 
 
 
