diff --git a/src/lib.rs b/src/lib.rs
index cada0dc..0d9d46e 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -104,12 +104,13 @@ This crate provides one cargo feature:
 #[doc(hidden)]
 pub mod lazy;
 
-#[cfg(test)]
-#[macro_use]
-extern crate doc_comment;
-
-#[cfg(test)]
-doctest!("../README.md");
+// ANDROID: disable tests that require doc_comment crate.
+//#[cfg(test)]
+//#[macro_use]
+//extern crate doc_comment;
+//
+//#[cfg(test)]
+//`doctest!("../README.md");
 
 #[cfg(feature = "spin_no_std")]
 #[path="core_lazy.rs"]
